### PR TITLE
Add mypy to the project

### PR DESCRIPTION
**Pull request by OmeGak**
Depends on:
- #10

### CODE REVIEW

File: src/unbeheader/typing.py
Comment (OmeGak) [inline src/unbeheader/typing.py:1]: Should this file belong in `stubs/unbeheader/__init__.py` instead?

Comment (OmeGak) [reply src/unbeheader/typing.py:1]: Probably not, since we are using it for runtime logic (i.e. `NamedTuple`).

File: src/unbeheader/__init__.py [hunk 1]
```diff
@@ -3,35 +3,32 @@
...
-    'sh': {
-        'regex': re.compile(r'((^#|[\r\n]#).*)*'),
-        'comments': {'comment_start': '#', 'comment_middle': '#', 'comment_end': ''}},
+from .typing import CommentSkeleton
+from .typing import SupportedFileType
+
+SUPPORTED_FILE_TYPES: dict[str, SupportedFileType] = {
+    'py': SupportedFileType(
+        re.compile(r'((^#|[\r\n]#).*)*'),
+        CommentSkeleton(comment_start='#', comment_middle='#', comment_end='')),
```
Comment (ThiefMaster) [inline src/unbeheader/__init__.py:12]: You can probably make this more concise by using a dataclass:

```python
@dataclass
class CommentSkeleton:
    start: str
    middle: str
    end: str = ''
```

`CommentSkeleton('#', '#')`
`CommentSkeleton('/*', ' *', ' */')`
etc.

Feel free to leave it as-is though.

**Review by ThiefMaster — APPROVED** [review]

File: src/unbeheader/__init__.py [hunk 2]
```diff
@@ -3,35 +3,32 @@
...
-    'sh': {
-        'regex': re.compile(r'((^#|[\r\n]#).*)*'),
-        'comments': {'comment_start': '#', 'comment_middle': '#', 'comment_end': ''}},
+from .typing import CommentSkeleton
+from .typing import SupportedFileType
+
+SUPPORTED_FILE_TYPES: dict[str, SupportedFileType] = {
+    'py': SupportedFileType(
+        re.compile(r'((^#|[\r\n]#).*)*'),
+        CommentSkeleton('#', '#', '')),
```
Comment (ThiefMaster) [inline src/unbeheader/__init__.py:12]:
```suggestion
        CommentSkeleton('#', '#')),
```

**Pull request merged**